src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs

```diff
@@ -30,17 +31,17 @@ public class ResponseCachingMiddleware
             RequestDelegate next,
             IOptions<ResponseCachingOptions> options,
             ILoggerFactory loggerFactory,
-            IResponseCachingPolicyProvider policyProvider,
-            IResponseCachingKeyProvider keyProvider)
+            ObjectPoolProvider poolProvider)
```

Comment:
Note that it's still necessary to resolve a ObjectPoolProvider from DI for the ResponseCachingKeyProvider (which was for perf). If/When we eventually create supported extension points, we'll need to mark this constructor as obsolete.